pkg/services/cloudmigration/api/api.go

```diff
@@ -32,25 +34,104 @@ func RegisterApi(
 }
 
 // RegisterAPIEndpoints Registers Endpoints on Grafana Router
-func (api *MigrationAPI) registerEndpoints() {
-	api.routeRegister.Group("/api/cloudmigrations", func(apiRoute routing.RouteRegister) {
-		apiRoute.Post(
-			"/migrate_datasources",
-			routing.Wrap(api.MigrateDatasources),
-		)
+func (cma *CloudMigrationAPI) registerEndpoints() {
+	cma.routeRegister.Group("/api/cloudmigration", func(cloudMigrationRoute routing.RouteRegister) {
+		// authentication
+		cloudMigrationRoute.Get("/authentication", routing.Wrap(cma.Authenticate))
+		cloudMigrationRoute.Post("/token", routing.Wrap(cma.CreateToken))
+		// migration
+		cloudMigrationRoute.Get("/migration", routing.Wrap(cma.GetMigrationList))
+		cloudMigrationRoute.Get("/migration/:id", routing.Wrap(cma.GetMigration))
+		cloudMigrationRoute.Post("/migration", routing.Wrap(cma.CreateMigration))
+		cloudMigrationRoute.Post("/migration/:id/run", routing.Wrap(cma.UpdateMigration))
+		cloudMigrationRoute.Post("/migration/:uid/run", routing.Wrap(cma.RunMigration))
```

Comment:
Maybe I am seeing it wrong, but won't these URLs collide?
Maybe change the first one to `/migration/:id/update`

Reply:
no, the first one should not have a run, it is just updating the migration, nothing to with the run